src/components/Page/Page.tsx

```diff
@@ -18,46 +14,26 @@ export interface PageProps extends HeaderProps {
   narrowWidth?: boolean;
 }
 
-type ComposedProps = PageProps & WithAppProviderProps;
-
-class PageInner extends React.PureComponent<ComposedProps, never> {
-  render() {
-    const {children, fullWidth, narrowWidth, ...rest} = this.props;
-
-    const className = classNames(
-      styles.Page,
-      fullWidth && styles.fullWidth,
-      narrowWidth && styles.narrowWidth,
-    );
-
-    const headerMarkup =
-      this.hasHeaderContent() === false ? null : <Header {...rest} />;
-
-    return (
-      <div className={className}>
-        {headerMarkup}
-        <div className={styles.Content}>{children}</div>
-      </div>
-    );
-  }
-
-  private hasHeaderContent(): boolean {
-    const {
-      title,
-      primaryAction,
-      secondaryActions,
-      actionGroups,
-      breadcrumbs,
-    } = this.props;
-
-    return (
-      (title != null && title !== '') ||
-      primaryAction != null ||
-      (secondaryActions != null && secondaryActions.length > 0) ||
-      (actionGroups != null && actionGroups.length > 0) ||
-      (breadcrumbs != null && breadcrumbs.length > 0)
-    );
-  }
+export function Page({children, fullWidth, narrowWidth, ...rest}: PageProps) {
+  const className = classNames(
+    styles.Page,
+    fullWidth && styles.fullWidth,
+    narrowWidth && styles.narrowWidth,
+  );
+
+  const hasHeaderContent =
+    (rest.title != null && rest.title !== '') ||
+    rest.primaryAction != null ||
+    (rest.secondaryActions != null && rest.secondaryActions.length > 0) ||
+    (rest.actionGroups != null && rest.actionGroups.length > 0) ||
+    (rest.breadcrumbs != null && rest.breadcrumbs.length > 0);
+
+  const headerMarkup = hasHeaderContent ? <Header {...rest} /> : null;
+
+  return (
+    <div className={className}>
+      {headerMarkup}
+      <div className={styles.Content}>{children}</div>
+    </div>
+  );
 }
-
-export const Page = withAppProvider<PageProps>()(PageInner);
```

Comment:
Was this even using AppProvider before?

Reply:
It was in v4. As part of ripping out app-bridge this component stopped using the `polaris` prop that withAppBridge provided, so yeah it not actually using what withAppProvider provided in the v5 branch.

I could have left this change as just removing the `withAppProvider` wrapping and having Page export a class component, but it was easy enough to migrate to a functional component so why not :)